app/actions/local/team.test.ts

```diff
@@ -0,0 +1,390 @@
+
+        const result = await removeUserFromTeam(serverUrl, teamId);
+
+        expect(getMyTeamById).toHaveBeenCalledWith(database, teamId);
+        expect(getTeamById).toHaveBeenCalledWith(database, myTeam.id);
+        expect(result.error).toEqual(new Error('Team not found'));
+    });
+
+    it('should handle errors and log them', async () => {
+        const error = new Error('Test error');
+        (getMyTeamById as jest.Mock).mockRejectedValue(error);
+
+        const result = await removeUserFromTeam(serverUrl, teamId);
+
+        expect(getMyTeamById).toHaveBeenCalledWith(database, teamId);
+        expect(logError).toHaveBeenCalledWith('Failed removeUserFromTeam', error);
+        expect(result).toEqual({error});
+    });
+
+    it('should handle when removeTeamFromTeamHistory returns undefined or empty array', async () => {
+        const myTeam = {id: 'myTeamId'};
+        const team = {id: 'teamId'};
+        const preparedModels = [
+            {id: 'model1', _preparedState: 'markAsDeleted'},
+            {id: 'model2', _preparedState: 'markAsDeleted'},
+        ];
+
+        (getMyTeamById as jest.Mock).mockResolvedValue(myTeam);
+        (getTeamById as jest.Mock).mockResolvedValue(team);
+        (prepareDeleteTeam as jest.Mock).mockResolvedValue(preparedModels);
+        (removeTeamFromTeamHistory as jest.Mock).mockResolvedValue(undefined);
+
+        const result = await removeUserFromTeam(serverUrl, teamId);
+
+        expect(getMyTeamById).toHaveBeenCalledWith(database, teamId);
+        expect(getTeamById).toHaveBeenCalledWith(database, myTeam.id);
+        expect(prepareDeleteTeam).toHaveBeenCalledWith(team);
+        expect(removeTeamFromTeamHistory).toHaveBeenCalledWith(operator, team.id, true);
+        expect(operator.batchRecords).toHaveBeenCalledWith(preparedModels, 'removeUserFromTeam');
+        expect(result).toEqual({error: undefined});
+    });
+
+    it('should handle when models length is 0', async () => {
+        const myTeam = {id: 'myTeamId'};
+        const team = {id: 'teamId'};
+        const models: never[] = [];
+
+        (getMyTeamById as jest.Mock).mockResolvedValue(myTeam);
+        (getTeamById as jest.Mock).mockResolvedValue(team);
+        (prepareDeleteTeam as jest.Mock).mockResolvedValue([]);
+        (removeTeamFromTeamHistory as jest.Mock).mockResolvedValue(models);
+
+        const result = await removeUserFromTeam(serverUrl, teamId);
+
+        expect(getMyTeamById).toHaveBeenCalledWith(database, teamId);
+        expect(getTeamById).toHaveBeenCalledWith(database, myTeam.id);
+        expect(prepareDeleteTeam).toHaveBeenCalledWith(team);
+        expect(removeTeamFromTeamHistory).toHaveBeenCalledWith(operator, team.id, true);
+        expect(operator.batchRecords).not.toHaveBeenCalled();
+        expect(result).toEqual({error: undefined});
+    });
+});
+
+describe('addSearchToTeamSearchHistory', () => {
+    const terms = 'search terms';
+    const MAX_TEAM_SEARCHES = 15;
```

Comment:
Would it be better to export MAX_TEAM_SEARCHES from ./local/team.ts and reuse it here vs creating two separate ones?

Maybe the idea is if we want to change MAX_TEAM_SEARCHES in implementation, the test will stay intact?

In the scenario a developer changing MAX_TEAM_SEARCHES to 20, then I don't think it will be any good to have the test failed unless there's a specific reason to tell the developer changing that value that it MUST be 15.

Reply:
this is Great point. I'll reuse the one declared in the team.ts. That way tests wont need change if the MAX_* value is changed.

Reply:
Got it Rahim.
I'll export the MAX_TEAM_SEARCHES and use it here